eskipfile/remote_test.go

```diff
@@ -82,18 +84,54 @@ func TestLoadAll(t *testing.T) {
 			Shunt:       false,
 			Backend:     "https://example.com/",
 		}},
+	}, {
+		title:           "Cached valid remote file",
+		routeContent:    "",
+		routeStatusCode: 304,
+		expected: []*eskip.Route{{
+			Id:   "VALID",
+			Path: "/",
+			Filters: []*eskip.Filter{{
+				Name: "setPath",
+				Args: []interface{}{
+					"/homepage/",
+				},
+			}},
+			BackendType: eskip.NetworkBackend,
+			Shunt:       false,
+			Backend:     "https://example.com/",
+		}},
+		cached: true,
 	},
 	} {
 		s := createTestServer(test.routeContent, test.routeStatusCode)
 		defer s.Close()
 
 		t.Run(test.title, func(t *testing.T) {
-			options := &RemoteWatchOptions{RemoteFile: s.URL, Threshold: 10, Verbose: true, FailOnStartup: true}
+			options := &RemoteWatchOptions{RemoteFile: s.URL, Threshold: 10, Verbose: true, FailOnStartup: true, EnableRoutesCaching: test.cached}
 			client, err := RemoteWatch(options)
 			if err == nil {
 				defer client.(*remoteEskipFile).Close()
 			}
 
+			if test.cached {
+				f, err := os.OpenFile(client.(*remoteEskipFile).localPath, os.O_APPEND|os.O_WRONLY, os.ModeAppend)
```

Comment:
The one that open(2) has also to do close(2).
So after error check `defer f.Close()`

And for readability, please use `fd` instead of `f` for file descriptors.